### DISCUSSION TITLE
Rewrite rule for root path "/"

### DISCUSSION THREAD

**Discussion by bledari** [vercel/next.js#61808] [Help] [answered]
### Summary

Locally I want "http://localhost:3000/" to transparently go to "example.com"

I can map subpaths successfully with the same rewrite feature, for example "http://localhost:3000/us" -> "example.com/us" ... etc, but for root it not clear what would the config be.

I am following this ("next": "14.1.0")
https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites

I need solution for v14.1.0 and v12.1.0
https://github.com/vercel/next.js/blob/v12.1.0/docs/api-reference/next.config.js/rewrites.md

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I tried with:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  async rewrites() {
    return {
      beforeFiles: [{ source: "/", destination: "https://example.com" }],
    };
  },
};

export default nextConfig;
```

And it works, the `/` goes to example, and other routes take me to the Next.js page.

**Reply by bledari** [reply]
It works, thank you!

**Reply by ghost** [reply]
Excuse me, why can't it be written like this here?

```ts
/** @type {import('next').NextConfig} */
const nextConfig = {
  async rewrites() {
    return [
      {
        source: "/",
        destination: "/emoji",
      },
    ];
  },
};

export default nextConfig;
```
